File: software/mosek/7/tools/examples/dotnet/concurrent2.cs
/*
 * Copyright: Copyright (c) MOSEK ApS, Denmark. All rights reserved.
 *
 * File:      concurrent2.cs
 *
 * Purpose:   To demonstrate a more flexible interface for concurrent optimization. 
*/

using System;

class msgclass : mosek.Stream 
{
  string prefix;
  public msgclass (string prfx) 
    {
      prefix = prfx;
    }
  
  public override void streamCB (string msg)
    {
      Console.Write ("{0}{1}", prefix,msg);
    }
}

public class concurrent1
{
  
  public static void Main (String[] args)
    {
      mosek.Env
        env = null;
      mosek.Task
        task = null;
      mosek.Task[]
        task_list = new mosek.Task[] { null };
            
      try
        {
          // Create mosek environment. 
          env  = new mosek.Env ();
          // Create a task object linked with the environment env.
          task = new mosek.Task (env, 0,0);
          // Directs the log task stream to the user specified
          // method task_msg_obj.streamCB
          task.set_Stream (mosek.streamtype.log, new msgclass ("simplex: "));

          task_list[0] = new mosek.Task (env, 0,0);
          task_list[0].set_Stream (mosek.streamtype.log, new msgclass ("intrpnt: "));

          task.readdata(args[0]);

          // Assign different parameter values to each task. 
          // In this case different optimizers. 
          task.putintparam(mosek.iparam.optimizer,
                           mosek.optimizertype.primal_simplex);
  
          task_list[0].putintparam(mosek.iparam.optimizer,
                                   mosek.optimizertype.intpnt);
  

          // Optimize task and task_list[0] in parallel.
          // The problem data i.e. C, A, etc. 
          // is copied from task to task_list[0].
          task.optimizeconcurrent(task_list);
          
          task.solutionsummary(mosek.streamtype.log);
        }
      catch (mosek.Exception e)
      {
        Console.WriteLine (e.Code);
        Console.WriteLine (e);
      }
          
      if (task != null) task.Dispose ();
      if (env  != null)  env.Dispose ();
          
    }
}

File: software/mosek/7/tools/examples/python/concurrent2.py
##
#  Copyright: Copyright (c) MOSEK ApS, Denmark. All rights reserved.
# 
#  File:      concurrent2.py
# 
#  Purpose:   To demonstrate a more flexible interface for concurrent optimization. 
##


import sys

import mosek

from mosek.array import array 

# Since the actual value of Infinity is ignores, we define it solely
# for symbolic purposes:
inf = 0.0

# Define a stream printer to grab output from MOSEK
class streamprinter:
    def __init__(self,prefix):
        self.prefix = str(prefix)
    def __call__(self,text):
        #sys.stdout.write (self.prefix + text)
        sys.stdout.write (self.prefix + text)
        sys.stdout.flush()
        pass

# We might write everything directly as a script, but it looks nicer
# to create a function.
def main (args):
    # Open MOSEK and create an environment and task
    # Create a MOSEK environment
    env = mosek.Env ()
    # Attach a printer to the environment
    env.set_Stream (mosek.streamtype.log, streamprinter("[env]"))

    # Create a task
    task = env.Task(0,0)
    # Attach a printer to the task
    task.set_Stream (mosek.streamtype.log, streamprinter("simplex: "))

    # Create a task
    task_list = [env.Task(0,0)]
    # Attach a printer to the task
    task_list[0].set_Stream(mosek.streamtype.log, streamprinter("intpnt: "))

    task.readdata(args[0]);

    # Assign different parameter values to each task. 
    # In this case different optimizers. 
    task.putintparam(mosek.iparam.optimizer,
                           mosek.optimizertype.primal_simplex)
  
    task_list[0].putintparam(mosek.iparam.optimizer,
                             mosek.optimizertype.intpnt)
    

    # Optimize task and task_list[0] in parallel.
    # The problem data i.e. C, A, etc. 
    # is copied from task to task_list[0].
    task.optimizeconcurrent(task_list)
    
    task.solutionsummary(mosek.streamtype.log)


# call the main function
try:
    main (sys.argv[1:])
except mosek.Exception as e:
    print ("ERROR: %s" % str(e.errno))
    if e.msg is not None:
        print ("\t%s" % e.msg)
    sys.exit(1)
except:
    import traceback
    traceback.print_exc()
    sys.exit(1)

 

